includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -712,7 +714,7 @@ private function process_payment_with_deferred_intent( int $order_id ) {
 
 			// Handle saving the payment method in the store.
 			// It's already attached to the Stripe customer at this point.
-			if ( $payment_information['save_payment_method_to_store'] ) {
+			if ( $payment_information['save_payment_method_to_store'] && $upe_payment_method && $upe_payment_method->get_id() === $upe_payment_method->get_retrievable_type() ) {
```

Comment:
This change can essentially be described as: only save the token if the payment method being used resolves to that same payment method. eg card = card 👍 bancontact = sepa ❌. 

When using a Bancontact to purchase a subscription we don't save the bancontact stripe payment method id (ie `pm_`) because it's not reusable. After the customer returns to the store after authorising the payment, stripe will create a `sepa_debit` token and we save it at that point. See `process_upe_redirect_payment()` and `process_order_for_confirmed_intent()`.